### ISSUE TITLE
used-before-assignment is not always triggered

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#267]
Originally reported by: **Claudiu Popa (BitBucket: [PCManticore](http://bitbucket.org/PCManticore), GitHub: @PCManticore)**

---

For the following code:

```py
class A(object):

    def __init__(self):

        self.a = True

    def method(self):

        if not self.a:
            test = 1
            dest = test + 2
        else:
            # UnboundLocalError: local variable 'test' referenced before assignment
            dest = test + 3
            test = 1


if __name__ == "__main__":
    a = A()
    a.method()

```

Maybe we can detect that `test` is conditionally defined.

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/267

**Comment by rogalski** [comment]
Should be labeled with `flow_control`

**Comment by Pierre-Sassoulas** [comment]
Closing as a duplicate of #1727, it's old but the discussion happened elsewhere. (Thank you @jacobtylerwalls for helping us cleanup)
